### DISCUSSION TITLE
Hide "input" from response upon 422 Validation Error

### DISCUSSION THREAD

**Discussion by themightywolfie** [fastapi/fastapi#10352] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import Body, FastAPI
from fastapi.security import HTTPBasicCredentials
from fastapi.responses import JSONResponse

app = FastAPI()

@app.post("/login")
async def login(credentials: HTTPBasicCredentials = Body(...)):
   return JSONResponse(content={'username': credentials.username})
```

### Description

The example code is a simplified version of the basic auth I am trying to implement. I am sending "username" and "password" in request body. 

If either of the fields are missing from the request body, I get the following detailed response because obviously the HTTPBasicCredentials pydantic model has only two fields. I used "user" instead of "username" in the request body

```
{
    "detail": [
        {
            "type": "missing",
            "loc": [
                "body",
                "username"
            ],
            "msg": "Field required",
            "input": {
                "user": "john.smith",
                "password": "3xtr3m#"
            },
            "url": "https://errors.pydantic.dev/2.4/v/missing"
        }
    ]
}
```
I want to hide the `input` from this response. Is there any way possible? 

### Operating System

Windows

### FastAPI Version

0.103.1

### Pydantic Version

2.4.0

### Python Version

3.11.4

**Comment by a-was** [comment]
Same here, but for the `url` field
Here's Pydantic docs on how to hide these fields
https://docs.pydantic.dev/2.4/api/pydantic_core/#pydantic_core._pydantic_core.ValidationError.json
Of course, FastAPI uses this method internally and should provide a way to set this kwargs

**Comment by NeilBotelho** [comment, marked as answer]
fastapi allows you to [override request validation exceptions](https://fastapi.tiangolo.com/tutorial/handling-errors/?h=handlin#override-request-validation-exceptions). One sort of solution to this is to override the exception with a modified version of the default fastapi [request validation exception handler](https://github.com/tiangolo/fastapi/blob/1bf5e7a10e33d6f65e0f701ec271785fbc9b463c/fastapi/exception_handlers.py#L20) and exclude the input/url field from the response i.e. :
```python
@app.exception_handler(RequestValidationError)
async def request_validation_exception_handler(
    request: Request, exc: RequestValidationError
) -> JSONResponse:
    print(exc.errors())
    return JSONResponse(
        status_code=HTTP_422_UNPROCESSABLE_ENTITY,
        content={"detail": jsonable_encoder(exc.errors(), exclude={"input"})})
```

The downside of this method would be that this removes the input/url field from all exception responses, unless theres a way to override the response exception handler for a single method.

**Reply by a-was** [reply]
Thanks, it works! 😃

**Reply by styper** [reply]
Shouldn't FastAPI hide URL and input by default since the OpenAPI generated docs doesn't have those fields? Or is my setup broken?
I used the tutorial example and passed in a string in place of the item_id and I get those fields, but the docs and redoc links don't reference them in the 422 error schema.

**Reply by kinuax** [reply]
Current FastAPI's validation error includes `input` field, not `url`. OpenAPI schema does not include `input` and `url`.

**Comment by kinuax** [comment]
I share another approach that proposes enabling/disabling both `input` and `url` fields without tweaking the request validation exceptions: https://github.com/fastapi/fastapi/pull/13657
